packages/streamr-logger/include/streamr-logger/StreamrLogFormatter.hpp

```diff
@@ -0,0 +1,175 @@
+
+struct LogLevelData {
+    folly::StringPiece logLevelName;
+    folly::StringPiece color;
+};
+
+// FATAL cannot be used in folly because it aborts. So CRITICAL is converted
+// to Streamr fatal
+constexpr LogLevelData getLogLevelData(const folly::LogLevel level) {
+    switch (level) {
+        case folly::LogLevel::DBG:
+            return {"TRACE", "\033[90m"};
+            break;
+        case folly::LogLevel::DBG0:
+            return {"DEBUG", "\033[34m"};
+            break;
+        case folly::LogLevel::INFO:
+            return {"INFO", "\033[32m"};
+            break;
+        case folly::LogLevel::WARN:
+            return {"WARN", "\033[33m"};
+            break;
+        case folly::LogLevel::ERR:
+            return {"ERROR", "\033[31m"};
+            break;
+        case folly::LogLevel::CRITICAL:
+            return {"FATAL", "\033[1;41m"};
+            break;
+        default:
+            return {"TRACE", "\033[90m"};
+            break;
+    }
+}
+
+} // namespace
+
+namespace streamr::logger {
+
+/*
+Format based on this Streamr log. The Filename and line number is fixed size
+with 36 characters. If it is longer then it is truncated
+DEBUG [2024-06-06T13:52:49.816] (<Filename>: <Line Number>     ):<Message>
+
+Examples of a truncated filename
+ERROR [2024-06-06T13:52:49.816] (12345678901234567890123456789012: 10):<Message>
+ERROR [2024-06-06T13:52:49.816] (1234567890123456789012345678901: 101):<Message>
+
+This one is not truncated:
+ERROR [2024-06-06T13:52:49.816] (1234567890123456.cpp: 101           ):<Message>
+
+DEBUG [2024-06-05T08:50:39.787] (File name): Message
+ERROR [2024-06-05T08:50:39.787] (File name): Message
+FATAL [2024-06-05T08:50:39.787] (File name): Message
+INFO [2024-06-05T08:50:39.787] (File name): Message
+TRACE [2024-06-05T08:50:39.787] (File name): Message
+WARN [2024-06-05T08:50:39.787] (File name): Message
+*/
+
+class StreamrLogFormatter : public folly::LogFormatter {
+public:
+    StreamrLogFormatter() = default;
+    ~StreamrLogFormatter() override = default;
+
+    struct StreamrLogMessage {
+        std::chrono::system_clock::time_point timestamp;
+        folly::StringPiece fileBasename;
+        unsigned int lineNumber;
+        folly::LogLevel logLevel;
+        const std::string& logMessage;
+    };
+
+    std::string formatMessageInStreamrStyle(const StreamrLogMessage& message) {
+        struct tm ltime;
+        const auto timeSinceEpoch = message.timestamp.time_since_epoch();
+        const auto epochSeconds =
+            std::chrono::duration_cast<std::chrono::seconds>(timeSinceEpoch);
+        const std::chrono::milliseconds millisecs =
+            std::chrono::duration_cast<std::chrono::milliseconds>(
+                timeSinceEpoch) -
+            epochSeconds;
+        time_t unixTimestamp = epochSeconds.count();
+        if (!localtime_r(&unixTimestamp, &ltime)) {
+            memset(&ltime, 0, sizeof(ltime));
```

Comment:
do this in more c++ way if there is one , memset looks quite low-level c code

Reply:
this was copy/paste from the folly code